crypto/include/internal/md32_common.h

```diff
@@ -93,6 +93,12 @@
 # error "HASH_BLOCK_DATA_ORDER must be defined!"
 #endif
 
+#ifdef __GNUC__
+typedef unsigned int ossl_uint __attribute((__aligned__(1)));
+#else
+typedef unsigned int ossl_uint;
+#endif
```

Comment:
I'd like to suggest a name that would reflect the attribute. As opposite to generic ossl_ prefix that is. misaligned_ comes to mind, but it's probably too long... Maybe char_as_uint would be appropriate?

Reply:
I always added "u" for unaligned.

Reply:
so if the base type is u32, I make it ossl_uu32.
ossl to avoid name space clashes, and the extra u means unaligned.

Reply:
and GETU32 for aligned memory but GETUU32 for unaligned memory...

Reply:
well, while I read "uint" here as unaligned int, but ossl_uuint would more be in line with
the other unaligned types?
So how about ossl_uuint ?

Reply:
ossl_uu32 would work.
I can drop the ossl_ prefix.

Reply:
maybe u_<type> ?  It's meaning is always a bit context sensitive, sometimes it is just
plain unaligned.
Sometimes it is aligned on strict alignment platforms and unaligned elsewhere.
so u_u64,  u_size_t etc?

Reply:
> `u_<type>`

Well, I for one don't find additional underscore as significant improvement. I mean it's still too "implied"... As already said the concern is readability. I mean occasional reader should rather ask him-/herself "how do they do that" than "what does it mean". From this angle e.g. `char_as_<type>` would be better choice. It would also better reflect what's going on platforms with strict alignment requirements. I mean on those platforms the cast is performed on *aligned* data. With this in mind e.g. `char_as_<type>_is_safe` would be even better, but it's too long... Maybe `safe_as_<type>`?

Reply:
Unaligned, maybe misaligned must be somewhere in the name.
When I add that attribute it should be even safe to use in armv4 with strict
alignment.

Reply:
But that is not intended, because the compiler will create 4 single byte moves.
That's not what I want.

Reply:
Well, or `una_<typ>` maybe `unaligned_<type>`  way longer than I would like,
I would say if we all can agree to that then it is probably worth fighting with the 80 column limit..